markets/bfp-market/contracts/modules/PerpAccountModule.sol

```diff
@@ -124,6 +141,130 @@ contract PerpAccountModule is IPerpAccountModule {
             );
     }
 
+    function splitAccount(
+        uint128 fromId,
+        uint128 toId,
+        uint128 marketId,
+        uint128 proportion
+    ) external {
+        Account.loadAccountAndValidatePermission(
+            fromId,
+            AccountRBAC._PERPS_MODIFY_COLLATERAL_PERMISSION
+        );
+        Account.loadAccountAndValidatePermission(
+            toId,
+            AccountRBAC._PERPS_MODIFY_COLLATERAL_PERMISSION
+        );
+        Runtime_splitAccount memory runtime;
+
+        PerpMarket.Data storage market = PerpMarket.exists(marketId);
+        Margin.Data storage toAccountMargin = Margin.load(toId, marketId);
+        Position.Data storage toPosition = market.positions[toId];
+
+        // Check that the to account is empty.
+        if (proportion > DecimalMath.UNIT) {
+            revert ErrorUtil.AccountSplitProportionTooLarge();
+        }
+        // Check that the to account is empty.
+        if (market.orders[toId].sizeDelta != 0 || market.orders[fromId].sizeDelta != 0) {
+            revert ErrorUtil.OrderFound();
+        }
+        if (toPosition.size != 0) {
+            revert ErrorUtil.PositionFound(toId, marketId);
+        }
+        // Note that we do not check for debt, as it's impossible for a trader to have debt with 0 collateral.
+        if (Margin.getCollateralUsdWithoutDiscount(toId, marketId) != 0) {
+            revert ErrorUtil.CollateralFound();
+        }
+
+        if (market.flaggedLiquidations[fromId] != address(0)) {
+            revert ErrorUtil.PositionFlagged();
+        }
+        runtime.oraclePrice = market.getOraclePrice();
+        PerpMarketConfiguration.Data storage marketConfig = PerpMarketConfiguration.load(marketId);
+        Position.Data storage fromPosition = market.positions[fromId];
+
+        // From account should not be liquidatable
+        if (
+            Position.isLiquidatable(
```

Comment:
Just thinking out loud but I wonder if the check below in L257 already implicitly checks this. Like, would it be possible to split a position that's just on the boarder of liquidation where the `toId` position cannot be liquidated?

Reply:
hmm good question, I think that could be possible, depending on how the market is configured

Reply:
I think let's keep it here to be on the safer side and optimise for gas later.